src/App.tsx

```diff
@@ -1,223 +1,294 @@
+import { TodoLIst } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
+import { Header } from './components/Header';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [errorAdd, setErrorAdd] = useState(false);
+  const [errorRemove, setErrorRemove] = useState(false);
+  const [errorUpdate, setErrorUpdate] = useState(false);
+  const [errorEmptyTitile, setErrorEmptyTitile] = useState(false);
+  const [typeFilter, setTypeFilter] = useState('All');
+  const [hidden, setHidden] = useState(true);
+  const [isAdding, setIsAdding] = useState<Todo | null>(null);
+  const [clearLoader, setClearLoader] = useState(false);
+  const [loadingAllTodos, setLoadingAllTodos] = useState(false);
+  const [isLoading, setIsLoading] = useState<Todo | null>(null);
+
+  const handleHidden = useCallback(
+    (value: boolean) => {
+      setHidden(value);
+    }, [],
+  );
+
+  const handleLoading = useCallback((value: Todo | null) => {
+    setIsLoading(value);
+  }, []);
+
+  const handleTypeFilter = useCallback((type: string) => {
+    setTypeFilter(type);
+  }, []);
+
+  const hanldeUpdateError = useCallback((value: boolean) => {
+    setErrorUpdate(value);
+  }, []);
+
+  const handlerRemoveError = useCallback((value: boolean) => {
+    setErrorRemove(value);
+  }, []);
+
+  const handlerNewTitle = useCallback((value: string) => {
+    setNewTodoTitle(value);
+  }, []);
+
+  const foundTodoList = useCallback((u:User) => {
+    getTodos(u.id).then(response => {
+      setTodoList(response);
+      setIsAdding(null);
+      setClearLoader(false);
+      setIsLoading(null);
+      setLoadingAllTodos(false);
+    });
+  }, [user]);
+
+  const createTodo = useCallback(
+    async (event:React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      if (!newTodoTitle.trim()) {
+        setErrorEmptyTitile(true);
+        setHidden(false);
+
+        return;
+      }
+
+      if (user) {
+        const newTodo = {
+          userId: user.id,
+          title: newTodoTitle,
+          completed: false,
+        };
+
+        setIsAdding(newTodo);
+
+        try {
+          await createTodos(newTodo);
+        } catch {
+          setErrorAdd(true);
+          setHidden(false);
+        }
+
+        foundTodoList(user);
+        setNewTodoTitle('');
+      }
+    }, [newTodoTitle],
+  );
+
+  const checkedAllCompletedTodo = (list: Todo[]) => (
+    list.every(todo => todo.completed)
+  );
+
+  const selectComplited = useCallback(async (todo:Todo) => {
+    try {
+      setIsLoading(todo);
+      if (todo.completed) {
+        await UpdateTodo(todo, false);
+      } else {
+        await UpdateTodo(todo, true);
+      }
```

Comment:
will be better to create separate variable with needed value of completed property, and call function for updating todo only once, to avoid code repeating